### ISSUE TITLE
Segmented Control crash if StackPanel Orientation value is Horizontal

### ISSUE THREAD

**Issue by WillyWoe** [CommunityToolkit/Labs-Windows#425]
### Describe the bug

If `StackPanel` Orientation value is set to Vertical it works, but if is set to Horizontal the app crashes:

**Xaml**

```xaml
<Window
    x:Class="LabsTests.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:LabsTests"
    xmlns:labs="using:CommunityToolkit.Labs.WinUI"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <StackPanel Orientation="Vertical"> <!--If Orientation value is Horizontal, the app crashes-->
        <labs:Segmented SelectedIndex="0"                        
                        SelectionMode="Single">
            <labs:SegmentedItem Content="Item 1">
                <labs:SegmentedItem.Icon>
                    <FontIcon Glyph="&#xEA3A;" />
                </labs:SegmentedItem.Icon>
            </labs:SegmentedItem>
            <labs:SegmentedItem Content="Item 2">
                <labs:SegmentedItem.Icon>
                    <FontIcon Glyph="&#xEA3A;" />
                </labs:SegmentedItem.Icon>
            </labs:SegmentedItem>
            <labs:SegmentedItem Content="Item 3 with a long label">
                <labs:SegmentedItem.Icon>
                    <FontIcon Glyph="&#xEA3A;" />
                </labs:SegmentedItem.Icon>
            </labs:SegmentedItem>
            <labs:SegmentedItem Content="Item 4">
                <labs:SegmentedItem.Icon>
                    <FontIcon Glyph="&#xEA3A;" />
                </labs:SegmentedItem.Icon>
            </labs:SegmentedItem>
        </labs:Segmented>
    </StackPanel>
</Window>
```

**NuGet references**

```csproj
<PackageReference Include="CommunityToolkit.Labs.WinUI.SegmentedControl" Version="0.0.2" />
<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.2.230118.102" />
<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.1" />
```

### Steps to reproduce

```text
1. Create a new WinUI Desktop project.
3. Copy the xml above and use the NuGet references.
4. Set `StackPanel` Orientation value to Horizontal in the example above.
5. Run the app.
```

### Expected behavior

Segmented Control should not crash if StackPanel Orientation is set to Horizontal.

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

Windows 10 22H2 Build 19045

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Visual Studio Version

2022

### Device form factor

Desktop

### Help us help you

No, I'm unable to contribute a solution.

**Comment by michael-hawker** [comment]
@WillyWoe I believe this is what will be fixed by #416? @niels9001 do you agree? Did we have an issue for this already, or should we just use this one?

**Comment by niels9001** [comment]
@WillyWoe Thanks for flagging this!

@michael-hawker Yep, just tested - and #416 resolves this bug. I think it's very similar (the width is unknown, hence the crash) so let's use this issue!
